### PR TITLE
feat(`check-values`): ensure `@variation` is a positive integer

### PR DESCRIPTION
**Pull request by brettz9**

### CODE REVIEW

File: .README/rules/check-values.md [hunk 1]
```diff
@@ -7,9 +7,11 @@ This rule checks the values for a handful of tags:
 2. `@since` - As with `@version`
 3. `@license` - Checks that there is a present and valid SPDX identifier
     or is present within an `allowedLicenses` option.
-4. `@author` - Checks there is a value present, and if the option
+4. `@author` - Checks that there is a value present, and if the option
     `allowedAuthors` is present, ensure that the author value is one
     of these array items.
+5. `@variation` - Checks that there iis a value present, and that it is
```
Comment (golopot) [inline .README/rules/check-values.md:13]: But the doc say you can use any value!

https://jsdoc.app/tags-variation.html

> You can provide any value with the @variation tag, as long as the combination of the value and the longname results in a globally unique version of the longname.

Comment (brettz9) [reply .README/rules/check-values.md:13]: Yes, I think you're right it should be optional and perhaps off by default, but it does specify `variationNumber` in the spec.

Comment (brettz9) [reply .README/rules/check-values.md]: Added now only if a non-default `numericOnlyVariation` is set to `true`

Comment (brettz9) [reply .README/rules/check-values.md]: Plan to be merging tonight, so please let me know if you are not done reviewing (or if you are done).

File: test/rules/assertions/checkValues.js [hunk 2]
```diff
@@ -206,6 +225,44 @@ export default {
...
+       * @variation
+       */
+      function quux (foo) {
+
+      }
+      `,
+      errors: [
+        {
+          line: 3,
+          message: 'Missing JSDoc @variation.',
```
Comment (golopot) [inline test/rules/assertions/checkValues.js:243]: Should be "Missing value for JSDoc variation."

Comment (brettz9) [reply test/rules/assertions/checkValues.js]: Thanks, fixed, along with the messages for other Missing tag values that preexisted.

**Pull request merged**